autokeras/hypermodel/preprocessor.py

```diff
@@ -673,3 +676,233 @@ def set_config(self, config):
         self.gaussian_noise = config['gaussian_noise']
         self.seed = config['seed']
         self.shape = config['shape']
+
+
+def return_zero():
+    return 0
+
+
+class FeatureEngineering(Preprocessor):
+    """A preprocessor block does feature engineering for the data.
+
+    # Arguments
+        column_types: A list of strings. The length of the list should be the same
```

Comment:
Passing a list that has the same length as the number of features in the data isn't a great API. It is very error prone.

Could it be feasible to pass a dict mapping column names (e.g. from a header) to their type? If a name is missing, its type could be automatically inferred from a sample data (which works in most cases): if all values in the sample data encode floats/ints, treat as numerical, else treat as categorical.

Reply:
Does tf.data.Dataset comes with column names or header? If so we can easily change, otherwise it would be hard for the pandas dataframe to reach in from fit function all the way to this block.